libsolidity/interface/CompilerStack.cpp

```diff
@@ -292,10 +292,12 @@ bool CompilerStack::analyze()
 					if (!resolver.resolveNamesAndTypes(*contract)) return false;
 					// Note that we now reference contracts by their fully qualified names, and
 					// thus contracts can only conflict if declared in the same source file.  This
-					// already causes a double-declaration error elsewhere, so we do not report
-					// an error here and instead silently drop any additional contracts we find.
-					if (m_contracts.find(contract->fullyQualifiedName()) == m_contracts.end())
-						m_contracts[contract->fullyQualifiedName()].contract = contract;
+					// should already cause a double-declaration error elsewhere.
+					solAssert(
```

Comment:
Maybe assert instead that there is an error?

Reply:
@chriseth with the member below?

Reply:
`solAssert(m_hasError, "");`
does not work

Reply:
`m_errorReporter.hasError()`

Reply:
doesn't work

Reply:
Does that mean, `// should already cause a double-declaration error elsewhere.` doesn't hold any more? Otherwise `m_errorReporter.hasError()` would not return false?

Reply:
I think what @chriseth meant was
```
if (m_contracts.find(contract->fullyQualifiedName()) == m_contracts.end()) 
    solAssert(m_errorReporter.hasErrors(), "");
m_contracts[contract->fullyQualifiedName()].contract = contract;
```

Reply:
Though even in the case an error was reported, I wouldn't override the contract. I'm still confused in what case could we have two equal contract fqdn's for valid inputs?

Reply:
Maybe because of remappings?